jetty-util/src/main/java/org/eclipse/jetty/util/TypeUtil.java

```diff
@@ -751,4 +756,52 @@ public T next()
             }
         };
     }
+
+    /**
+     * Used on a {@link ServiceLoader#stream()} with {@link Stream#flatMap(Function)},
+     * so that in the case a {@link ServiceConfigurationError} is thrown it warns and
+     * continues iterating through the service loader.
+     * <br>Usage Example:
+     * <p>{@code ServiceLoader.load(Service.class).stream().flatMap(TypeUtil::providerMap).collect(Collectors.toList());}</p>
+     * @param <T> The class of the service type.
+     * @param provider The service provider to instantiate.
+     * @return a stream of the loaded service providers.
+     */
+    public static <T> Stream<T> mapToService(ServiceLoader.Provider<T> provider)
+    {
+        try
+        {
+            return Stream.of(provider.get());
+        }
+        catch (ServiceConfigurationError error)
+        {
+            LOG.warn("Service Provider failed to load", error);
+            return Stream.empty();
+        }
+    }
+
+    /**
+     * Shortcut method combining {@link #serviceProviderStream(ServiceLoader)} with
+     * with {@link #mapToService(ServiceLoader.Provider)} using {@link Stream#flatMap(Function)}.
+     * @param serviceLoader the ServiceLoader instance to use.
+     * @param <T> the type of the service to load.
+     * @return a stream of the service provider type which will not throw {@link ServiceConfigurationError}.
+     */
+    public static <T> Stream<T> serviceStream(ServiceLoader<T> serviceLoader)
```

Comment:
Can we have this be the only public API?
Make the other 2 APIs be private (or even inlined into this method)